geopy/geocoders/here.py

```diff
@@ -0,0 +1,281 @@
+        https://developer.here.com/documentation/geocoder/
+    """
+
+    structured_query_params = {
+        'city',
+        'county',
+        'district',
+        'country',
+        'state',
+        'street',
+        'housenumber',
+        'postalcode',
+    }
+
+    def __init__(
+            self,
+            app_id,
+            app_code,
+            format_string=None,
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            ssl_context=DEFAULT_SENTINEL,
+    ):
+        """
+
+        :param str app_id: Should be a valid HERE Maps APP ID
+            See https://developer.here.com/authenticationpage.
+
+        :param str app_code: Should be a valid HERE Maps APP CODE
+            See https://developer.here.com/authenticationpage.
+
+        :param str format_string:
+            See :attr:`geopy.geocoders.options.default_format_string`.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+        """
+        super(Here, self).__init__(
+            format_string=format_string,
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+        )
+        self.app_id = app_id
+        self.app_code = app_code
+        self.api = "%s://geocoder.api.here.com/6.2/geocode.json" % self.scheme
+        self.reverse_api = "%s://reverse.geocoder.api.here.com/6.2/reversegeocode.json" \
+            % self.scheme
+        # FIXME(?): add multi-reverse geocode
+        # FIXME(?): add landmark geocode
+
+    def geocode(
+            self,
+            query,
+            exactly_one=True,
+            language=None,
+            additional_data=False,
+            other=None,
+            timeout=DEFAULT_SENTINEL
+    ):
+        """
+        Return a location point by address.
+
+        Documentation at:
+            https://developer.here.com/documentation/geocoder/topics/resource-geocode.html
+
+        :param str query: The address or query you wish to geocode.
+
+            For a structured query, provide a dictionary whose keys
+            are one of: `city`, `county`, `district`, `country`, `state`,
+            `street`, `housenumber`, or `postalcode`.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
```

Comment:
This is out of order. `timeout` kwarg is the last one, this doc should be too.

Reply:
Ok, sure.